fairscale/experimental/nn/ssd_offload.py

```diff
@@ -68,6 +69,20 @@ def read(input_tensor: torch.Tensor, filename: str, file_offset_bytes: int = 0)
             assert data_read == chunk_end - chunk_start
 
 
+class StorageState(Enum):
+    """
+    Simple enum to indicate whether the tensor handle is pointing
+    to data on disk or memory. This is useful for asserting on 
+    whether the tensor is available for operations or if it needs
+    to be moved from disk to CPU or device.
+    """
+
+    UNALLOCATED = auto()
+    ON_DISK = auto()
+    ON_CPU = auto()
+    ON_GPU = auto()
```

Comment:
(related to my CUDA stream remark): I don't see this being used actually, is that planned in a follow up or is the expectation that it will be done somewhere else on the stack (by the user who consumes this) ?

Reply:
good point! We don't use it right now and we probably won't since this is the only class that should be handing this state. Initially I was contemplating sharing this structure with the FSDP codebase. Will remove it for now.